GetStream/stream-chat-swift
# Fix wrong message text font when text accessibility is larger + Snapshot Testing Framework Update

### 🔗 Issue Links
Resolves https://github.com/GetStream/ios-issues-tracking/issues/371

### 🎯 Goal
Fixes the message text font when the text accessibility setting is set to larger. 

This issue was introduced by this [PR](https://github.com/GetStream/stream-chat-swift/pull/2328), which we tried to fix the dynamic font size for the message text, but this actually causes the text font to be with an incorrect size after closing/opening the app. So we need to revert that change. 

For now, we don't know the root cause of why the font is not changed dynamically, and it only changes after closing the app. But there is a workaround to overcome this, and it is to override the `MessageContentView.defaultMessageFont` to always return a new UIFont instance (Ex: `UIFont.preferredFont(forTextStyle: .body)`).  More context [here](https://twitter.com/nunovieira_dev/status/1578440760361644032?s=20).

**Bonus:** Fixes the message timestamp label and checkmark delivery status issues when the font size is larger. 

### 📝 Summary
- Fixes the message text font when the text accessibility setting is set to larger.
- Fixes the timestamp label being clipped when the text font size is larger.
- Fixes the checkmark in the delivery status view being too small for larger text font sizes.
- Updates the Snapshot Testing framework to improve M1 and Intel SSs

### 🎨 Showcase

| Before | After |
| ------ | ----- |
|  <img src="https://user-images.githubusercontent.com/12814114/233067166-a90aa79a-3699-4ede-bc18-81c465bf9ef3.png"/>   |  <img src="https://user-images.githubusercontent.com/12814114/233067238-a9728bef-8edf-45a7-9554-5bac19b368d8.png"/>  |

### 🧪 Manual Testing Notes

1. Xcode > Open Developer Tool > Accessibility Inspector
2. Processes > Simulator
3. Tap on Settings button (Right top corner)
4. Set the Dynamic font to a really big font
5. Close and open the app
6. Open a channel
7. The font of the messages should have the exact size of the Input Placeholder

### ☑️ Contributor Checklist
